src/core/Form/Combobox/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,595 @@
+      });
+    };
+
+    const filter = (data: ComboboxData, query: string) =>
+      data.labelText.toLowerCase().includes(query.toLowerCase());
+
+    const handleKeyDown = (event: React.KeyboardEvent) => {
+      const { filteredItems: items, currentSelection } = this.state;
+      const index = items.findIndex(
+        ({ uniqueItemId }) => uniqueItemId === currentSelection,
+      );
+
+      const getNextIndex = () => (index + 1) % items.length;
+      const getPreviousIndex = () => (index - 1 + items.length) % items.length;
+
+      const getNextItem = () => items[getNextIndex()];
+      const getPreviousItem = () => items[getPreviousIndex()];
+
+      const scrollItemList = (lblTxt: string) => {
+        if (
+          this.popoverListRef !== null &&
+          this.popoverListRef.current !== null &&
+          showPopover
+        ) {
+          const idOfCurrentElement = `${id}-${lblTxt}`;
+          const elementOffsetTop =
+            document.getElementById(idOfCurrentElement)?.offsetTop || 0;
+          const elementOffsetHeight =
+            document.getElementById(idOfCurrentElement)?.offsetHeight || 0;
+          if (elementOffsetTop < this.popoverListRef.current.scrollTop) {
+            this.popoverListRef.current.scrollTop = elementOffsetTop;
+          } else {
+            const offsetBottom = elementOffsetTop + elementOffsetHeight;
+            const scrollBottom =
+              this.popoverListRef.current.scrollTop +
+              this.popoverListRef.current.offsetHeight;
+            if (offsetBottom > scrollBottom) {
+              this.popoverListRef.current.scrollTop =
+                offsetBottom - this.popoverListRef.current.offsetHeight;
+            }
+          }
+        }
+      };
+
+      switch (event.key) {
+        case 'ArrowDown': {
+          event.preventDefault();
+          focusToMenu();
+          const nextItem = getNextItem();
+          if (nextItem) {
+            this.setState({ currentSelection: nextItem.uniqueItemId });
+            scrollItemList(nextItem.uniqueItemId);
+          }
+          break;
+        }
+
+        case 'ArrowUp': {
+          event.preventDefault();
+          focusToMenu();
+          const previousItem = getPreviousItem();
+          if (previousItem) {
+            this.setState({ currentSelection: previousItem.uniqueItemId });
+            scrollItemList(previousItem.uniqueItemId);
+          }
+          break;
+        }
+
+        case 'Enter': {
+          event.preventDefault();
+          if (currentSelection) {
+            const currentItem = items.find(
+              ({ uniqueItemId }) => uniqueItemId === currentSelection,
+            );
+            if (currentItem) {
+              this.handleItemSelection(currentItem);
+            }
+          }
+          break;
+        }
+
+        case 'Escape': {
+          event.preventDefault();
+          this.setState(
+            (
+              _: ComboboxState<T & ComboboxData>,
```

Comment:
It seems there is no proper convention for dismissing unused parameters and destructuring cannot be used here I assume. Maybe name the parameter `_parameterName` so that it's more clear what parameter it is in addition to it not being used.

Reply:
This is now renamed to `_prevState`